src/base/kaldi-error.cc

```diff
@@ -131,70 +130,64 @@ std::string KaldiGetStackTrace() {
 
 void KaldiAssertFailure_(const char *func, const char *file,
                          int32 line, const char *cond_str) {
-  std::ostringstream ss;
-  ss << "KALDI_ASSERT: at " << GetProgramName() << func << ':'
-     << GetShortFileName(file)
-     << ':' << line << ", failed: " << cond_str << '\n';
+  MessageLogger ml(LogMessageEnvelope::Error, func, file, line);
+  ml.stream() << "Asserion failed: " << cond_str;
 #ifdef HAVE_EXECINFO_H
-  ss << "Stack trace is:\n" << KaldiGetStackTrace();
+  ml.stream() << "\nStack trace is:\n" << KaldiGetStackTrace();
 #endif
-  std::cerr << ss.str();
-  std::cerr.flush();
-  // We used to call abort() here, but switch to throwing an exception
-  // (like KALDI_ERR) because it's easier to deal with in multi-threaded
-  // code.
-  throw std::runtime_error(ss.str());
 }
 
-
-KaldiWarnMessage::KaldiWarnMessage(const char *func, const char *file,
-                                   int32 line) {
-  this->stream() << "WARNING (" << GetProgramName() << func << "():"
-                 << GetShortFileName(file) << ':' << line << ") ";
+LogHandler SetLogHandler(LogHandler new_handler) {
+  LogHandler old_handler = g_log_handler;
+  g_log_handler = new_handler;
+  return old_handler;
 }
 
-KaldiWarnMessage::~KaldiWarnMessage() {
-  std::string str = ss_.str();
-  while (!str.empty() && str[str.length() - 1] == '\n')
-    str.resize(str.length() - 1);
-  fprintf(stderr, "%s\n", str.c_str());
-}
-
-
-KaldiLogMessage::KaldiLogMessage(const char *func, const char *file,
-                                 int32 line) {
-  this->stream() << "LOG (" << GetProgramName() << func << "():"
-                 << GetShortFileName(file) << ':' << line << ") ";
-}
-
-KaldiLogMessage::~KaldiLogMessage() {
-  std::string str = ss_.str();
-  while (!str.empty() && str[str.length() - 1] == '\n')
-    str.resize(str.length() - 1);
-  fprintf(stderr, "%s\n", str.c_str());
-}
+static void SendToLog(const LogMessageEnvelope& envelope,
```

Comment:
we tend to put the & to the right, next to the variable name, in situations like this...